markdown.lua

```diff
@@ -1,227 +1,191 @@
+local function getNameRaceAgeProf(unit)
+    --%s is a placeholder for a string, and %d is a placeholder for a number.
+    return string.format("%s, %d years old %s.", dfhack.units.getReadableName(unit), df.global.cur_year - unit.birth_year, dfhack.units.getProfessionName(unit))
 end
 
-if flerb == 'textviewer' then
-    local scrn = scrn --as:df.viewscreen_textviewerst
-
-    local lines = scrn.src_text
-
-    if lines ~= nil then
-
-        local log = getFileHandle()
-        log:write('### ' .. dfhack.df2utf(scrn.title) .. '\n')
+-- Main logic for item and unit processing
+local item = dfhack.gui.getSelectedItem(true)
+local unit = dfhack.gui.getSelectedUnit(true)
 
-        print('Exporting ' .. dfhack.df2console(scrn.title) .. '\n')
+if not item and not unit then
+    print([[
+Error: No unit or item is currently selected.
+- To select a unit, click on it.
+- For items that are installed as buildings (like statues or beds), 
+open the building's interface in the game and click the magnifying glass icon.
 
-        for n,x in ipairs(lines) do
-            log:write(reformat(dfhack.df2utf(x.value)).." ")
--- debug output
---            print(x.value)
-        end
-        closeFileHandle(log)
-    end
-
-elseif flerb == 'announcelist' then
-    local scrn = scrn --as:df.viewscreen_announcelistst
-
-    local lines = scrn.reports
-
-    if lines ~= nil then
-        local log = getFileHandle()
-        local lastTime = ""
-
-        for n,x in ipairs(lines) do
-            local currentTime = formattime(x.year, x.time)
-            if (currentTime ~= lastTime) then
-                lastTime = currentTime
-                log:write('\n***\n\n')
-                log:write('## ' .. currentTime .. '\n')
-            end
--- debug output
---            print(x.text)
-            log:write(x.text .. '\n')
-        end
-        closeFileHandle(log)
-    end
-
-
-elseif flerb == 'topicmeeting' then
-    local lines = scrn.text --hint:df.viewscreen_topicmeetingst
-
-    if lines ~= nil then
-        local log = getFileHandle()
+Please select a valid target in the game and try running the script again.]])
+    -- Early return to avoid proceeding further
+    return
+end
 
-        for n,x in ipairs(lines) do
--- debug output
---            print(x.value)
-            log:write(x.value .. '\n')
+local log = getFileHandle()
+
+if item then
+    -- Item processing
+    local item_raw_name = dfhack.items.getDescription(item, 0, true)
+    local item_raw_description = df.global.game.main_interface.view_sheets.raw_description
+    log:write('### ' .. dfhack.df2utf(item_raw_name) .. '\n\n#### Description: \n' .. reformat(dfhack.df2utf(item_raw_description)) .. '\n')
+    print('Exporting description of the ' .. item_raw_name)
+
+elseif unit then   
+    -- Unit processing
+    -- Simulate UI interactions to load data into memory (click through tabs)
+    local screen = dfhack.gui.getDFViewscreen()
+    -- Click "Personality" 
+    df.global.gps.mouse_x = 145
```

Comment:
The x coordinates will depend on the screen resolution since the view sheets panel moves relative to the right screen edge. Subtract a constant from the width returned from dfhack.screen.getWindowSize()

Reply:
Done, and tested on Linux machine with other resolution